spec/routes/client_spec.rb

```diff
@@ -1,23 +1,106 @@
-require File.expand_path '../../spec_helper.rb', __FILE__
+require_relative '../spec_helper.rb'
 
-describe 'The Client list endpoint' do
+describe 'The Clients endpoint' do
   def app
     Iam
   end
 
   include Rack::Test::Methods
 
   before(:all) do
-    FactoryGirl.create(:client, name: 'Client X')
+    # anything that should happen before all tests
   end
 
   it 'responds OK' do
     get '/clients'
     expect(last_response).to be_ok
   end
 
-  it 'contains the name of a client we added' do
-    get '/clients'
-    expect(last_response).to match(/Client X/)
+  it 'includes the names of existing clients' do
+    FactoryGirl.create(:client, name: 'Client X')
+    FactoryGirl.create(:client, name: 'Client Y')
+    FactoryGirl.create(:client, name: 'Client Z')
+
+    get 'clients'
+
+    expect(last_response.body).to include('Client X')
+    expect(last_response.body).to include('Client Y')
+    expect(last_response.body).to include('Client Z')
+  end
+
+  it 'displays a specific client by id' do
+    client = Client.create(name: 'New Client')
+    get "/clients/#{client.id}"
+    expect(last_response.body).to include('New Client')
+  end
+
+  it 'answers 404 when asked for a non-existent client' do
+    get '/clients/1200438'
+    expect(last_response.status).to eq(404)
+  end
+
+  it 'responds ok when asked for clients/new' do
+    get '/clients/new'
+    expect(last_response.status).to eq(200)
+  end
+
+  it 'responds ok when asked for the form to edit an existing client' do
+    client = Client.create(name: 'Editable')
+    get "/clients/#{client.id}/edit"
+    expect(last_response.status).to eq(200)
+  end
+
+  it 'responds 404 when asked for the form to edit an absent client' do
+    get '/clients/9884093/edit'
+    expect(last_response.status).to eq(404)
+  end
+
+  it 'allows us to create a new client, then redirects to the list' do
+    post '/clients', name: 'I\'m new!'
```

Comment:
I was under the impression that any string with `'` was to be double quoted; `"I'm new!"` for instance. Escaping characters is pretty ugly imo.